src/fake-timers-src.js

```diff
@@ -450,7 +450,7 @@ function withGlobal(_global) {
 
                 // ensures identity checks using the constructor prop still works
                 // this should have no other functional effect
-                this.constructor = Date;
+                this.constructor = NativeDate;
```

Comment:
Good catch, this was naively working in the tests due to it being the same when not overwritten. This is what I _meant_ to have written 😄

Reply:
👍 I was initially very surprised, but makes sense now — since this happens every time a fake Date is created.

Reply:
patch version out shortly